lib/shanty_template_plugin/template_plugin.rb

```diff
@@ -0,0 +1,85 @@
+require 'deep_merge'
+require 'erubis'
+require 'fileutils'
+require 'shanty/plugin'
+require 'yaml'
+
+module ShantyTemplatePlugin
+  # Public: Shanty template plugin
+  class TemplatePlugin < Shanty::Plugin
+    PROJECT_EXTENSION = 'erb'
+    DS_EXTENSTION = 'syaml'
+
+    option :datasource, 'plain'
+    option :outputdir, 'build'
+    subscribe :build, :on_build
+    projects "**/*.#{PROJECT_EXTENSION}"
+
+    def on_build(project)
+      load_data(project).each do |env, data|
+        project_files(project).each do |template|
+          output = output_filename(project, template, env)
+          FileUtils.mkdir_p(File.dirname(output))
+          File.open(output, 'w') do |out|
+            out.write(Erubis::Eruby.new(File.read(template)).result(data: data))
+          end
+        end
+      end
+    end
+
+    def artifacts(project)
+      load_data(project).keys.flat_map do |env|
+        project_files(project).map do |template|
+          filename = output_filename(project, template, env)
+          Shanty::Artifact.new(File.extname(filename), self.class.name, URI("file://#{filename}"))
+        end
+      end
+    end
+
+    private
+
+    def output_filename(project, template, env)
+      file = File.basename(template).split('.')
+      file.delete(PROJECT_EXTENSION)
```

Comment:
You can privide a suffix to strip when calling `File.basename`:

```ruby
File.basename(template, PROJECT_EXTENSION).split('.')
```

You then don't have to call `#delete`.